openmrs/omod/src/main/webapp/profiles.jsp

```diff
@@ -2,104 +2,155 @@
+        </span>
+    </form>
 </div>
 
 <div class="section profile-select">
-  <c:choose>
-    <c:when test="${empty profiles}">
-      No profiles are available.
-    </c:when>
-    <c:otherwise>
-      <form method="post">
-        Select a profile:
-        <select name="profile" id="profile-select" ${not authorized ? "disabled" : ""}
-            size="${fn:length(profiles) < 3 ? 3 : fn:length(profiles)}">
-          <c:forEach var="file" items="${profiles}">
-            <fmt:formatDate value="${file.modified}" pattern="MMM dd, HH:mm" var="formattedDate"/>
-            <c:set var="formattedLine" value="${formattedDate} | ${fn:escapeXml(file.formattedName)} | ${file.formattedSize}"/>
-            <option value="${fn:escapeXml(file.name)}" class="filename"
-                    ${file.name == currentProfile ? 'selected' : ''}>
-              ${fn:replace(formattedLine, ' ', '&nbsp;')}
-            </option>
-          </c:forEach>
-        </select>
-        &nbsp;&nbsp;&nbsp;
-        <input type="submit" name="op" value="Apply" ${not authorized ? "disabled" : ""}>
-        <input type="submit" name="op" value="Download" ${not authorized ? "disabled" : ""}>
-        <input type="submit" name="op" value="Delete" ${not authorized ? "disabled" : ""}
-            onclick="s = document.getElementsByTagName('select')[0]; return confirm('Delete ' + s.options[s.selectedIndex].value + '?')">
-      </form>
-    </c:otherwise>
-  </c:choose>
+    <c:choose>
+        <c:when test="${empty profiles}">
+            No profiles are available.
+        </c:when>
+        <c:otherwise>
+            <form method="post" id="profile-form">
+                Select a profile:
+                <select name="profile" id="profile-select" size="3">
+                    <c:forEach var="file" items="${profiles}">
+                        <fmt:formatDate value="${file.modified}" pattern="MMM dd, HH:mm" var="formattedDate"/>
+                        <c:set var="formattedLine" value="${formattedDate} | ${fn:escapeXml(file.formattedName)} | ${file.formattedSize}"/>
+                        <option value="${fn:escapeXml(file.name)}" class="filename" ${file.name == currentProfile ? 'selected' : ''}>
+                            ${fn:replace(formattedLine, ' ', '&nbsp;')}
+                        </option>
+                    </c:forEach>
+                </select>
+                &nbsp;&nbsp;&nbsp;
+                <input type="submit" name="op" value="Apply">
+                <input type="submit" name="op" value="Download">
+                <input type="submit" name="op" value="Delete">
+            </form>
+        </c:otherwise>
+    </c:choose>
 </div>
 
+<script type="text/javascript">
+    (function() {
+        var profileQtd = ${fn:length(profiles)};
+        if(profileQtd > 3){
+            document.forms["profile-form"]["profile"].size = profileQtd;
+        }
+
+        var authorized = ${authorized};
+        if (!authorized) {
```

Comment:
This works, but it mixes Java, JSTL EL, and JavaScript, relying on the fact that the Java boolean is converted to a string by JSTL and the Java boolean has a toString() that produces "true" or "false", which coincidentally happens to be spelled the same as a JavaScript boolean... is that what you intended?  It causes JavaScript to be emitted into the page even when it isn't used, so I'm wondering if you mean to use ${authorized} in the JSTL way, e.g. 
```
    <c:if test="${not authorized}">
      <script>
        for (var i = 0; i < document.forms.length; i++) {
          ...
        }
      </script>
    </c:if>
```